src/utils/vaapi.rs

```diff
@@ -692,6 +762,9 @@ where
     display: Rc<Display>,
     /// The metadata state. Updated whenever the decoder reads new data from the stream.
     pub(crate) metadata_state: StreamMetadataState,
+    /// Whether we support reusing old contexts after DRC. This depends on the
+    /// codec in use.
+    supports_context_reuse: bool,
```

Comment:
I do not know, as I've said.

I tried one example Nicolas sent me, but there's another issue: there's something broken with our code for odd resolutions, somehow. These always fail.

My hunch though is that yes, we can probably reuse this for VP8 and H.264 as well, as we do not regress any fluster tests

Reply:
What I've found out so far: asking ccdec to output NV12 works, i.e. at least some frames with odd resolutions are displayed fine. There's more corruption later, but this is probably due to something else I think

The GStreamer element (vavp8dec) can only output NV12, which then gets converted to I420 using using some SIMD sorcery and then that gets further converted through `videocodectestsink` .

@nicolas since you wrote `videocodectestsink`, do you happen to know any limitations/known workarounds regarding odd-resolutions?

Reply:
This is what we get using I420: 
[ccdec4-i420.tar.gz](https://github.com/chromeos/cros-codecs/files/11630710/ccdec4-i420.tar.gz)

Versus what we get using NV12
[ccdec4-nv12.tar.gz](https://github.com/chromeos/cros-codecs/files/11630716/ccdec4-nv12.tar.gz)

Both resolutions are 212x173

Reply:
Hi there, GStreamer software processing (like orc code) is a bit like HW, it needs things to have a certain alignment. At the very basic, all odd width or height get rounded up to be even. With subsampling, we also round up luma to the next multiple of 4, this is for pre-fetching. This padding is not included into the ITU conformance test MD5 sum.

Instead of converting in videocodectestsink, what we do is that we compute an MD5SUM from the displayed pixels, skipping over the padding at the end of each lines. For NV12, do Y, and read all U from the second plane, and then read all V from the second plane, just to avoid a bit of coversion (but not yet integrated into fluster).